setuptools/config/expand.py

```diff
@@ -0,0 +1,330 @@
+"""Utility functions to expand configuration directives or special values
```

Comment:
This file was originally extracted from the existing implementation of `setuptools.config` and after that some improvements were added (e.g. allowing explicitly specifying a project root directory)